sdk/core/azure-core/src/main/java/com/azure/core/implementation/SwaggerInterfaceParser.java

```diff
@@ -47,15 +49,16 @@ public SwaggerInterfaceParser(Class<?> swaggerInterface, SerializerAdapter seria
             if (hostAnnotation != null && !hostAnnotation.value().isEmpty()) {
                 this.host = hostAnnotation.value();
             } else {
-                throw new MissingRequiredAnnotationException(Host.class, swaggerInterface);
+                throw logger.logExceptionAsError(new MissingRequiredAnnotationException(Host.class, swaggerInterface));
```

Comment:
These logs are more for us than a customer right? They wouldn't be able to do anything for this.

@JonathanGiles would we want debugging only logging?

Reply:
Yeah, customers should hopefully never see this, as it would mean we've shipped something that didn't work. I'm ok with it being an Error, so that we definitely see it during development.

Reply:
Revert it back.